analyses/cell-type-neuroblastoma-04/scripts/00b_convert-nbatlas.R

```diff
@@ -77,51 +75,54 @@ all_cell_ids <- readr::read_lines(opts$cell_id_file)
 # keep only cells that are present in in `all_ids`
 nbatlas_seurat <- subset(nbatlas_seurat, cells = all_cell_ids)
 
-# convert Seurat to SCE object directly, to save space in the final object
-nbatlas_sce <- SingleCellExperiment(
-  assays = list(
-    counts = nbatlas_seurat[["RNA"]]$counts,
-    logcounts = nbatlas_seurat[["RNA"]]$data
-  )
-)
+# if testing, subset to fewer cells: keep 5% of each label
+if (opts$testing) {
+  keep_cells <- nbatlas_seurat@meta.data |>
+    tibble::rownames_to_column("cell_id") |>
+    dplyr::group_by(Cell_type) |>
+    dplyr::sample_frac(0.05) |>
+    dplyr::pull(cell_id)
 
-# add in colData, including updated cell labels with `neuroendocrine-tumor` label for tumor cells
-colData(nbatlas_sce) <- nbatlas_seurat@meta.data |>
-  dplyr::mutate(
-    cell_id = rownames(colData(nbatlas_sce)),
-    NBAtlas_label = ifelse(
-      cell_id %in% tumor_cells,
-      "Neuroendocrine-tumor",
-      Cell_type
-    )
-  ) |>
-  DataFrame(row.names = rownames(colData(nbatlas_sce)))
+  nbatlas_seurat <- subset(nbatlas_seurat, cells = keep_cells)
+}
+
+# convert to SCE, using `as` to avoid CI error
+# note that Seurat gives some deprecation warnings, but this works
+nbatlas_sce <- as.SingleCellExperiment(nbatlas_seurat)
```

Comment:
Did you check to make sure we don't need to do any renaming of assays or anything after you do this? The docs don't note what the format is after conversion... https://satijalab.org/seurat/reference/as.singlecellexperiment